plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -284,16 +284,16 @@ getIdeas nfp = do
                      Just <$> liftIO (parseModuleEx flags' fp contents')
 
         setExtensions flags = do
-          hlintExts <- getExtensions flags nfp
+          hlintExts <- getExtensions nfp
           debugm $ "hlint:getIdeas:setExtensions:" ++ show hlintExts
           return $ flags { enabledExtensions = hlintExts }
 
-getExtensions :: ParseFlags -> NormalizedFilePath -> Action [Extension]
-getExtensions pflags nfp = do
+getExtensions :: NormalizedFilePath -> Action [Extension]
+getExtensions nfp = do
     dflags <- getFlags
     let hscExts = EnumSet.toList (extensionFlags dflags)
     let hscExts' = mapMaybe (GhclibParserEx.readExtension . show) hscExts
-    let hlintExts = nub $ enabledExtensions pflags ++ hscExts'
+    let hlintExts = nub hscExts'
```

Comment:
Without looking at what line 295 does in detail, can `nub` be removed? Seems like the addition of `enabledExtensions pflags` was the reason for removing dupes.

Reply:
Yeah I thought about removing it but thought "what if hscExts' also has dupes". However if I'm thinking that then I should think that about everything. So will remove it.

Reply:
I'm just presuming `EnumSet` has proper `Set` semantics of no duplicates, and that `readExtension` doesn't map different strings to the same extension. I could be wrong though.

Reply:
you're totally right